packages/shared/src/hooks/useConditionalFeature.ts

```diff
@@ -24,11 +24,19 @@ export const useConditionalFeature = <T extends JSONValue>({
   const { user } = useContext(AuthContext);
   const { growthbook } = useGrowthBookContext();
   const { ready } = useFeaturesReadyContext();
-  const queryKey = generateQueryKey(RequestKey.Feature, user, feature.id);
+  const queryKey = generateQueryKey(
+    RequestKey.Feature,
+    user,
+    feature.id,
+    shouldEvaluate,
+  );
 
   const { data: featureValue, isLoading } = useQuery(
     queryKey,
     () => {
+      if (!shouldEvaluate) {
+        return false as WidenPrimitives<T>;
+      }
```

Comment:
What about features that are not bool?

Reply:
I can change it to the default value, would that make more sense I guess?

Reply:
Yes, just to support other feature flags in this hook.

Reply:
@capJavert Although, at some point we might change default to true, and then you get this experiment forever.
But guess once we hit that we can add something like `valueWhenNotEvaluated` or something.

Reply:
Hmm, don't know at top of my head, but sounds complicated and yes something to evaluate for future usage of the hook.

Reply:
Basically just returning something that GB does not return feels wrong right? We should hide the added UI in regards to something else and feature value should always be true if user is part of experiment.

Reply:
Yeah correct, for this flow it's fine, but can imagine there might be something where it won't work ilke this.
There we could potentially use the initial evaluation for rendering determination.